### PR TITLE
FYLE-2a81g3v-1: Platform changes - Tax groups service

### PR DESCRIPTION
**Pull request by Dimple16**
### Code coverage report:
Screenshot 2022-06-21 at 9 46 47 PM

### CODE REVIEW

File: src/app/core/services/tax-group.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import { TaxGroupService } from './tax-group.service';
+
+const taxGroupPlatformResponse = {
+  count: 2,
+  data: [
+    {
+      id: 'tgwfjbgqo32',
+      org_id: 'orwruogwnngg',
+      created_at: new Date('2020-06-01T13:14:54.804+00:00'),
+      updated_at: new Date('2020-06-11T13:14:55.201598+00:00'),
```
Comment (Julias0) [inline src/app/core/services/tax-group.service.spec.ts:13]: ok the response dosnt return a date object na - this needs to happen inside the transform step, you basically pass the value you get from the response into new Date

File: src/app/core/services/tax-group.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    name: 'VAT',
+    percentage: 20,
+    created_at: new Date('2020-06-01T13:14:54.804+00:00'),
+    updated_at: new Date('2020-06-11T13:14:55.201598+00:00'),
+    org_id: 'orwruogwnngg',
+    is_enabled: true,
+  },
+];
+
+describe('TaxGroupService', () => {
```
Comment (Julias0) [inline src/app/core/services/tax-group.service.spec.ts:54]: Please add a screenshot of the code coverage report for this file - `tax-group.service.ts`
We should be maintaining 100% code coverage

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comments to the unit tests

File: src/app/core/services/tax-group.service.spec.ts (hunk 2 above)
Comment (Dimple16) [reply src/app/core/services/tax-group.service.spec.ts:61]: will share the screenshot on thread

**Review by Julias0 — APPROVED** [review]

File: src/app/core/services/tax-group.service.spec.ts (hunk 2 above)
Comment (Dimple16) [reply src/app/core/services/tax-group.service.spec.ts:91]: added screenshot

File: src/app/core/services/tax-group.service.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+export class TaxGroupService {
+  constructor(
+    @Inject(PAGINATION_SIZE) public paginationSize: number,
+    private spenderPlatformApiService: SpenderPlatformApiService
+  ) {}
+
+  get(): Observable<TaxGroup[]> {
+    return this.getEnabledTaxGroupsCount().pipe(
+      switchMap((count) => {
+        count = count >= this.paginationSize ? count / this.paginationSize : 1;
```
Comment (Dimple16) [inline src/app/core/services/tax-group.service.ts:21]: @Julias0 changed this from `count > this.paginationSize` -> `count >= this.paginationSize` 
without this, covering the test case code `count / this.paginationSize` wouldn't be possible even after the pagination size change

**Review by Julias0 — APPROVED** [review]

**Pull request merged**